mlflow/pyfunc/utils/data_validation.py

```diff
@@ -34,6 +35,7 @@ class FuncInfo(NamedTuple):
     input_param_name: str
 
 
+@track_api_usage
```

Comment:
Can we fix the type hint of `track_api_usage`?

<img width="498" height="242" alt="Image" src="https://github.com/user-attachments/assets/6120b3e6-d806-4ea5-923d-1df6f4411014" />

Reply:
You can check checkout master and hover on `log_model`.

Reply:
Thanks, seems wrapt can solve this, let me update

Reply:
Ohh seems `cast` works, lemme update

Reply:
It does. Does this happen to other functions?